geode-lucene/src/test/java/org/apache/geode/cache/lucene/LuceneQueriesReindexDUnitTest.java

```diff
@@ -251,27 +203,42 @@ public void recreateDifferentIndexesWithDifferentFieldsShouldPass(
     executeTextSearch(accessor);
   }
 
-  private void waitAndVerifyThatIndexCreationFails(AsyncInvocation ai1, AsyncInvocation ai2) {
-    try {
-      Awaitility.await().atMost(1, TimeUnit.MINUTES).until(() -> {
-        boolean status1;
-        boolean status2;
-        try {
-          status1 = ai1.getException() instanceof UnsupportedOperationException;
-          assertTrue(status1);
-        } catch (AssertionError er) {
-          try {
-            status2 = ai2.getException() instanceof UnsupportedOperationException;
-            assertTrue(status2);
-          } catch (AssertionError er2) {
-            assertTrue(false);
-          }
-        }
+  private void verifyCreateIndexWithDifferentFieldShouldFail() throws InterruptedException {
+    AsyncInvocation ai1 = dataStore1.invokeAsync(() -> {
+      createIndex("text");
+    });
+
+    AsyncInvocation ai2 = dataStore2.invokeAsync(() -> {
+      createIndex("text2");
+    });
+
+    // wait for 10 seconds first for threads to finish
+    ai1.join(10000);
+    ai2.join(10000);
+
+    // if one thread is still alive after 10 seconds, will try to create the same index on the
+    // other datastore to unblock the thread.
+    if (ai1.isAlive()) {
+      dataStore2.invoke(() -> {
+        createIndex("text");
       });
-    } finally {
-      ai1.cancel(true);
-      ai2.cancel(true);
     }
+    if (ai2.isAlive()) {
+      dataStore1.invoke(() -> {
+        createIndex("text2");
+      });
+    }
+
+    // wait again for maximum 60 seconds for threads to die.
+    ai1.join(60000);
+    ai2.join(60000);
+
+    // at this time, both threads should be dead already, then we check the exception status
+    // if either thread is still alive, the getException call will throw an exception which would
+    // make the test fail.
+    boolean ai1HasException = ai1.getException() instanceof UnsupportedOperationException;
+    boolean ai2HasException = ai2.getException() instanceof UnsupportedOperationException;
+    assertThat(ai1HasException || ai2HasException).isTrue();
```

Comment:
This test may result in false positives:
1. a1 starts create index
2. a2 starts create index with invalid parameters.
3. assuming in future a mistake was made and a2 does not send out an exception and dies.
4. we see that a2 is dead and we start create index with a1's parameter in a2.
5. again assume if a1 throws an exception because of some code issue at this point.
6. The test will still pass as a1 has an exception but it was not expected.

In my opinion, the exception should be checked immediately after the isAlive() checks are done. As after re-starting the create index operations exceptions may be thrown for reasons other than invalid parameters.